### PR TITLE
Use qtpy instead of rolling custom loader

### PR DESCRIPTION
**Pull request by hmaarrfk**
Use qtpy for Qt backend selector

I think I also happened to kill support for PySide (1). Is it time to drop support for 2.7???

closes: https://github.com/jupyter/qtconsole/issues/279

### CODE REVIEW

File: qtconsole/qt.py [hunk 1]
```diff
@@ -1,23 +1,5 @@
 """ A Qt API selector that can be used to switch between PyQt4/5 and PySide.
```
Comment (ccordoba12) [inline qtconsole/qt.py]: I think this submodule should be removed and `qtpy` used directly wherever is needed.

**Comment by ccordoba12** [comment]
> I think I also happened to kill support for PySide (1)

`qtpy` still supports it.

> Is it time to drop support for 2.7???

Not yet, but in any case it shouldn't be part of this PR.

**Comment by hmaarrfk** [comment]
i don't really have a plan for getting past the 2.7 test fail.

Do you know it is happening?

File: qtconsole/frontend_widget.py [hunk 2]
```diff
@@ -169,19 +168,10 @@ def _lexer_default(self):
 
     def __init__(self, local_kernel=_local_kernel, *args, **kw):
         super(FrontendWidget, self).__init__(*args, **kw)
-        # FIXME: remove this when PySide min version is updated past 1.0.7
-        # forcefully disable calltips if PySide is < 1.0.7, because they crash
```
Comment (hmaarrfk) [inline qtconsole/frontend_widget.py]: @ccordoba12 this is why I think I killed PySide support.

Comment (hmaarrfk) [reply qtconsole/frontend_widget.py]: I guess it is reasonable to expect people not to be using a version from 2011??

https://pypi.org/project/PySide/

**Comment by hmaarrfk** [comment]
Is the `test_debug` known to be inconsistent? 

I'm getting inconsistent results locally

https://travis-ci.org/jupyter/qtconsole/jobs/588225647#L611

One set failed on python 3.6, while the other failed on python 2.7.

**Comment by ITAYC0HEN** [comment]
Looks good. Rebase is required

**Comment by hmaarrfk** [comment]
Lets see what happens.

**Comment by ITAYC0HEN** [comment]
Looks green :)

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -11,7 +11,7 @@ Use Conda to install dependencies and activate the development environment.
 ```
 conda create -n qtdev python=3
 conda activate qtdev
-conda install pyqt
```
Comment (ccordoba12) [inline CONTRIBUTING.md]: `qtpy` doesn't bring `pyqt`, so this needs to be

    conda install pyqt qtpy

File: examples/inprocess_qtconsole.py [hunk 4]
```diff
@@ -10,7 +10,7 @@
 """
 
 
-from qtconsole.qt import QtGui
+from qtpy import QtGui
```
Comment (ccordoba12) [inline examples/inprocess_qtconsole.py]: This needs to be

    from qtpy import QtWidgets

to use the right import below.

File: qtconsole/frontend_widget.py (hunk 2 above)
Comment (ccordoba12) [reply qtconsole/frontend_widget.py]: Agreed.

**Comment by hmaarrfk** [comment]
Thanks, hopefully all passes

**Comment by ccordoba12** [comment]
> Is it time to drop support for 2.7???

It is. But let's release this in the last Python 2 compatible version. Afterwards you could send us a Python 3 migration PR

**Comment by ITAYC0HEN** [comment]
Congratulations! Good work, folks :)

**Comment by ccordoba12** [comment]
I'll try to release this during the next week.

**Comment by hmaarrfk** [comment]
this, using qtpy as a unified backend for my software is definitely really nice!

**Comment by ITAYC0HEN** [comment]
> I'll try to release this during the next week.

Hey @ccordoba12! :) Can you please release upstream to pypi so we can use it from pip? :)

**Comment by ccordoba12** [comment]
Done.

**Pull request merged**